### PR TITLE
tests: new workflow step used to report spread error to mongodb

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The change includes:
 . New tool to report errors to mongodb
. New workflow step used to collect spread errors and invoque the new tool

The mongo host and port are going to be set in the environment of the agent runners

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -380,3 +393,16 @@ jobs:
...
+        if [ -e spread.log ]; then
+            echo "Running spread log analyzer"
+            ./tests/lib/tools/log-parser spread.log --print-detail none --print-results failed --output spread-results.json --cut 100
+            while IFS= read -r line; do
+                echo "Reporting spread error..."
+                ./tests/lib/tools/report-mongodb --db-name snapd --db-collection spread_errors "$line"
+            done <<< $(jq -cr '.[] | select( .type == "info") | select( .info_type == "Error")' spread-results.json)
+        else
+            echo "No spread log found, skipping errors reporting"
+        fi
```
Comment (stolowski) [inline .github/workflows/test.yaml:408]: Nitpick, missing newline.

File: tests/lib/tools/report-mongodb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        print("report-mongodb: invalid db collection, it does not exist in db")
+        sys.exit(1)
+
+    json_value = _get_json(value)
+    if not json_value:
+        print("report-mongodb: invalid value, it does not contain a valid json format")
+        sys.exit(1)
+
+    db[col].insert_one(json_value)
+    print("report-mongodb: json value inserted in db")
```
Comment (stolowski) [inline tests/lib/tools/report-mongodb:57]: Could you paste an example of the json output we would be inserting into mongodb?

Also, for someone unfamiliar with mongodb, can you explain how is this json blob going to be treated by mongodb and if later will we be able to sum up failures of same tests?

**Review by stolowski — COMMENTED** [review]
This looks very nice! But one question since I'm not familiar with mongodb.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply .github/workflows/test.yaml]: added

File: tests/lib/tools/report-mongodb (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/tools/report-mongodb]: Hi, I just added an example of the data which is being sent. This is just a tool to report, then we will have another tool to retrieve data with aggregations, but initially, the tool is just reporting errors to mongo which stores json values in a collection called spread_errors which is contained in the snapd database. 
The idea is to allow in the future, sending another info to the database using a different collection.

**Review by mardy — APPROVED** [review]
LGTM!

File: tests/lib/tools/report-mongodb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        #    "verb": "executing",
+        #    "task": "google:ubuntu-18.04-64:tests/main/auto-refresh:parallel",
+        #    "extra": null,
+        #    "detail": {
+        #        "lines": ["-----\n",
+        #                  "snapd-state: expected a new auto-refresh change with id greater than 1, but it didn't happen\n",
+        #                  "-----\n",
+        #                  ".\n"]
+        #    }
+        # }
```
Comment (stolowski) [inline tests/lib/tools/report-mongodb:68]: Thanks for adding this!

**Review by stolowski — APPROVED** [review]
LGTM, thank you!

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -371,7 +386,7 @@ jobs:
           echo "::add-matcher::.github/spread-problem-matcher.json"
           export NESTED_BUILD_SNAPD_FROM_CURRENT=true
           export NESTED_ENABLE_KVM=true
-          spread -abend google-nested:${{ matrix.system }}:tests/nested/...
+          spread -abend google-nested:${{ matrix.system }}:tests/nested/... | tee spread.log
```
Comment (anonymouse64) [inline .github/workflows/test.yaml:389]: this again broke spread tests reporting failing spread tests in GitHub Actions as being successful

**Pull request merged**